panel/lxqtpanel.cpp

```diff
@@ -1226,25 +1234,41 @@ void LXQtPanel::userRequestForDeletion()
     emit deletedByUser(this);
 }
 
-void LXQtPanel::showPanel(bool animate)
+void LXQtPanel::showPanel()
 {
     if (mHidable)
     {
         mHideTimer.stop();
+
         if (mHidden)
-        {
-            mHidden = false;
-            setPanelGeometry(mAnimationTime > 0 && animate);
-        }
+            mShowTimer.start();
+    }
+}
+
+void LXQtPanel::showPanelWork()
+{
+    if (geometry().contains(QCursor::pos()))
```

Comment:
Please, describe the reasoning of adding this check... as by simple testing I don't see any difference in behaviour with or without the `geometry().contains(..` check.

If the check is not necessary, we can get rid of the whole `showPanelWork()` and use:
```diff
diff --git a/panel/lxqtpanel.cpp b/panel/lxqtpanel.cpp
index 535d971..cb7c272 100644
--- a/panel/lxqtpanel.cpp
+++ b/panel/lxqtpanel.cpp
@@ -210,7 +210,7 @@ LXQtPanel::LXQtPanel(const QString &configGroup, LXQt::Settings *settings, QWidg
 
     mShowTimer.setSingleShot(true);
     mShowTimer.setInterval(mShowDelay);
-    connect(&mShowTimer, SIGNAL(timeout()), this, SLOT(showPanelWork()));
+    connect(&mShowTimer, &QTimer::timeout, [this] { doShowPanel(true); });
 
     ensureVisible();

 ```